### PR TITLE
Arm backend: Add Ethos-U FVP tests for MLPerf Tiny models

### PR DESCRIPTION
**Pull request by tirwu01**
Add model definitions and Arm backend tests for four MLPerf Tiny benchmark models: ResNet8, DS-CNN, Deep AutoEncoder, and MobileNetV1-0.25.

Model definitions are placed under examples/models/mlperf_tiny/. Each model has tests for tosa_FP, tosa_INT, u55_INT and u85_INT pipelines in backends/arm/test/models/.

Notable model adaptations for Arm delegation:
- Deep AutoEncoder: Fuse Linear + BatchNorm1d pairs before export since the TOSA quantizer only annotates conv + batch_norm patterns.
- DS-CNN: Replace AvgPool2d(24, 5) with AdaptiveAvgPool2d(1) to satisfy the Ethos-U55 stride <= 3 constraint; the DecomposeAdaptiveAvgPool2dPass decomposes it into stride-1 pools.

Change-Id: I8dbf5e8a4b80996faab9f850c21740899f6b36fd

### Summary
[PLEASE REMOVE] See [CONTRIBUTING.md's Pull Requests](https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests) for ExecuTorch PR guidelines.

[PLEASE REMOVE] If this PR closes an issue, please add a `Fixes #<issue-id>` line.

[PLEASE REMOVE] If this PR introduces a fix or feature that should be the upcoming release notes, please add a "Release notes: <area>" label. For a list of available release notes labels, check out [CONTRIBUTING.md's Pull Requests](https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests).

### Test plan
[PLEASE REMOVE] How did you test this PR? Please write down any manual commands you used and note down tests that you have written if applicable.

cc @digantdesai @freddan80 @per @zingo @oscarandersson8218 @mansnils @Sebastian-Larsson @robell

### CODE REVIEW

**Comment by zingo** [comment]
Hi @digantdesai @rascani and @psiddh this adds a few new mpu nice models to the example folder and need a Meta review 🙏 🙂

**Review by rascani — APPROVED** [review]

File: backends/arm/test/models/test_mobilenet_v1_025.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+model_inputs = _wrapper.get_example_inputs()
+input_t = Tuple[torch.Tensor]
+
+quant_test_data = {
+    "per_channel_quantization=true": True,
+    "per_channel_quantization=false": False,
+}
+
+
+def test_mobilenet_v1_025_tosa_FP():
```
Comment (digantdesai) [inline backends/arm/test/models/test_mobilenet_v1_025.py:34]: Do they add any more coverage besides what we have for Mobilenet? Let's just add them only in examples if they aren't too different from what we already have, rationale is the CI job freq

File: backends/arm/test/models/test_ds_cnn.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+model_inputs = _wrapper.get_example_inputs()
+input_t = Tuple[torch.Tensor]
+
+quant_test_data = {
+    "per_channel_quantization=true": True,
+    "per_channel_quantization=false": False,
+}
+
+
+def test_ds_cnn_tosa_FP():
```
Comment (digantdesai) [inline backends/arm/test/models/test_ds_cnn.py:38]: same here as MV1

File: backends/arm/test/models/test_deep_autoencoder.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+model_inputs = _wrapper.get_example_inputs()
+input_t = Tuple[torch.Tensor]
+
+quant_test_data = {
+    "per_channel_quantization=true": True,
+    "per_channel_quantization=false": False,
+}
+
+
+def test_deep_autoencoder_tosa_FP():
```
Comment (digantdesai) [inline backends/arm/test/models/test_deep_autoencoder.py:64]: same here as MV1

File: backends/arm/test/models/test_resnet8.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+model_inputs = _wrapper.get_example_inputs()
+input_t = Tuple[torch.Tensor]
+
+quant_test_data = {
+    "per_channel_quantization=true": True,
+    "per_channel_quantization=false": False,
+}
+
+
+def test_resnet8_tosa_FP():
```
Comment (digantdesai) [inline backends/arm/test/models/test_resnet8.py:32]: same here as MV1

File: backends/arm/test/models/test_mobilenet_v1_025.py (hunk 1 above)
Comment (tirwu01) [reply backends/arm/test/models/test_mobilenet_v1_025.py:34]: Hi, MobileNetV1-0.25 is a distinct model from MobileNetV2/V3 — it's the specific architecture used in the MLPerf Tiny.These four models (ResNet8, DS-CNN, Deep AutoEncoder, MobileNetV1-0.25) are the standard MLPerf Tiny benchmark suite and are tested together as a set.
